### ISSUE TITLE
I can't seem to get anything out of this handler

### ISSUE THREAD

**Issue by corbesero** [zuazo/chef_handler_sns-cookbook#3]
I am sure this is user error on my part, but I am trying to follow the README.  I see the handler get installed.  I even see an entry in the chef-client log that the handler is being called.  But I see no SNS messages begin sent.    I have added a subject and body template.  

I don't know what else to try.

**Comment by zuazo** [comment]
Please, provide the code snippet you are using and the Chef log output.

Keep in mind that if you want the handler to send reports when there are no errors, you will need to add `:report` to the `node['chef_handler_sns']['supports']` attribute of this cookbook.

**Comment by corbesero** [comment]
I was just about to include all that stuff, but then I found an error in one of the logs.  \
[2016-04-28T18:52:40+00:00] ERROR: Running exception handlers
[2016-04-28T18:52:40+00:00] ERROR: Report handler Chef::Handler::Sns raised #<Chef::Exceptions::ValidationFailed: Required argument **access_key** is missi
ng!>

That doesn't come out in the normal output.

Anyway.  I don't have an access key.  I am using iam roles.  Shouldn't it be falling back to the iam role ?

**Comment by zuazo** [comment]
You are right, it should try to use IAM roles by default. Can you paste your `ohai` command output?

**Comment by corbesero** [comment]
i removed the /etc/passwd and group stuff.  its still pretty big .

[ohai.txt](https://github.com/zuazo/chef_handler_sns-cookbook/files/241362/ohai.txt)

**Comment by corbesero** [comment]
fixed.   the iam.json hint must be enabled.

**Comment by lavanyaruth** [comment]
Error executing action `enable` on resource 'chef_handler[Chef::Handler::Sns]'
      NoMethodError
      -------------
      undefined method `new' for ChefHandler:Module

  Iam new to chef, can anyone please explain me what makes this error to occur?

**Comment by zuazo** [comment]
@lavanyaruth this is an old closed issue. Please, don't add a comment here. Open a new issue if you need it.
